packages/dds/matrix/src/matrix.ts

```diff
@@ -594,19 +744,65 @@ export class SharedMatrix<T = any>
 				new ObjectStoragePartition(storage, SnapshotPath.cols),
 				this.serializer,
 			);
-			const [cellData, pendingCliSeqData] = await deserializeBlob(
-				storage,
-				SnapshotPath.cells,
-				this.serializer,
-			);
+			const [
+				cellData,
+				pendingCliSeqData,
+				cellLastWriteTracker,
+				setCellPolicySwitchOpSeqNumber,
+			] = await deserializeBlob(storage, SnapshotPath.cells, this.serializer);
 
 			this.cells = SparseArray2D.load(cellData);
 			this.pending = SparseArray2D.load(pendingCliSeqData);
+			// Old summaries will not have this info, so check that.
+			if (cellLastWriteTracker !== undefined) {
+				assert(
+					setCellPolicySwitchOpSeqNumber !== undefined,
+					"Both tracker and seq number should be present",
+				);
+				this.cellLastWriteTracker = SparseArray2D.load(cellLastWriteTracker);
+				this.setCellPolicySwitchOpSeqNumber = setCellPolicySwitchOpSeqNumber;
+			}
 		} catch (error) {
 			this.logger.sendErrorEvent({ eventName: "MatrixLoadFailed" }, error);
 		}
 	}
 
+	/**
+	 * Tells whether the setCell op should be applied or not based on First Write Win policy.
+	 */
+	private shouldSetCellBasedOnFWW(
+		rowHandle: Handle,
+		colHandle: Handle,
+		message: ISequencedDocumentMessage,
+	) {
+		assert(message.clientId !== null, "clientId should not be null");
```

Comment:
Should this functon assert that isSetCellOpMadeInFWWPolicy(message) is true?

Reply:
I don't want to put to that dependency here. It assumes that we are in FWW mode. Caller should make sure that is the case.

Reply:
Mentioned that in the comment.